### ISSUE TITLE
Python installation not detected by Quarto

### ISSUE THREAD

**Issue by desautm** [quarto-dev/quarto-cli#3500]
### Bug description

I'm on Windows 10 and Quarto can't seem to detect my Python installation.

```
Nom de l’hôte:                              L108235
Nom du système d’exploitation:              Microsoft Windows 10 Entreprise
Version du système:                         10.0.19044 N/A build 19044
Fabricant du système d’exploitation:        Microsoft Corporation
```

```
C:\Users\mdesautels>py --list
 -V:3.11 *        Python 3.11 (64-bit)
 -V:3.9
 -V:ContinuumAnalytics/Anaconda39-64 Anaconda py39_4.12.0

C:\Users\mdesautels>quarto check

[>] Checking Quarto installation......OK
      Version: 1.2.269
      Path: C:\Users\mdesautels\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.2.1
      Path: C:/ProgrammesMAD/R-4.2.1
      LibPaths:
        - C:/ProgrammesMAD/R-4.2.1/library
      rmarkdown: 2.18

[>] Checking Knitr engine render......OK
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
"py" instead of "python" is strange to me, although I'm not a windows user. I suspect there's some issue in your configuration that isn't standard. @cderv?

**Comment by cderv** [comment]
> "py" instead of "python" is strange to me, although I'm not a windows user.

@cscheid  `py` is a specific tool for Python for windows. It is called [Python Launcher ](https://docs.python.org/3/using/windows.html#python-launcher-for-windows) and we recommend it currently on Windows 
https://quarto.org/docs/computations/python.html#installation

We should even use to detect the version of Python active and configured

@desautm I'll try to reproduce. I am not using the Py launcher myself and I have python in PATH (I am using [pyenv-win](https://github.com/pyenv-win/pyenv-win) to manage my python versions )

Workaround for you to make Quarto work - Set `QUARTO_PYTHON` to the python version you need so that it skip auto detection and use the version provided.

**Comment by cderv** [comment]
@desautm Do you have `PY_PYTHON` set ? and if so to which value ? 

Type in terminal powershell 
````powershell
$env:PY_PYTHON
```

**Comment by cderv** [comment]
So I have been experimenting in a clean environment using Windows Sandbox. 

It may come to an interaction between having a conda version and non conda version. Is you anaconda version working ? 

Our current detection process is working as described in https://github.com/quarto-dev/quarto-cli/issues/1463 (which I believe could be improved). 

* py launcher will be used on Windows first only if `PY_PYTHON` is set, which I believe it is not by default. If you set this environment variable it may solve your issue. 
* When this env var is empty, the PATH will be searched. 
   * First conda will be check as it provides only `python` binary. 
   * Then `python3` binary
 
First one found is used. `QUARTO_PYTHON` overrides this and force the version to use. 

So can you check the following in addition to `py --list`: What is seen in your PATH ? 
In powershell: 
````powershell
(gcm python).Source
(gcm python3).Source
````

I think depending on the result it will help us understand what is happening. I can't reproduce on my side because installing python for windows from python.org puts a `python` in the PATH and Quarto founds it - even if I have Conda also installed. 

We need your help on this to check if you are in an edge case we don't yet support. Thank you

**Comment by aalexandersson** [comment]
I am also using Python 3.11 on Windows 10 installed with `pyenv-win` but no Python Launcher. Quarto (version 1.1.189) works on my C: drive (not on my network drives, including F:), and I have the same problem with `quarto check` not recognizing the Python 3 installation:

```
C:\temp>quarto check

[>] Checking Quarto installation......OK
      Version: 1.1.189
      Path: C:\Program Files\RStudio\bin\quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

(-) Checking R installation...........Warning message:
Project requested R version '4.0.2' but '4.2.0' is currently being used
[>] Checking R installation...........OK
      Version: 4.2.0
      Path: C:/PROGRA~1/R/R-4.2.0
      LibPaths:
        - C:/temp/renv/library/R-4.2/x86_64-w64-mingw32
        - C:/Users/aalexandersson/AppData/Local/Temp/RtmpwdCEB9/renv-system-library
      rmarkdown: (None)

      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
```
@cderv To answer your question, I get this output in Powershell:

```
PS F:\> (gcm python).Source
C:\Users\aalexandersson\.pyenv\pyenv-win\shims\python.bat
PS F:\> (gcm python3).Source
C:\Users\aalexandersson\.pyenv\pyenv-win\shims\python3.bat
```

Also, why do I get the warning message? Why is R version `4.0.2` requested?
```
Project requested R version '4.0.2' but '4.2.0' is currently being used
```

**Comment by cderv** [comment]
> I am also using Python 3.11 on Windows 10 installed with pyenv-win

This is key info, thanks ! This means your issue is a duplicated of mine https://github.com/quarto-dev/quarto-cli/issues/1463 that I opened because of this exactly. 

Look at this issue for context, but currently 

* Quarto does use `py.exe` to find python on Windows
* `pyenv-win` is then not supported unfortunately for auto detection by quarto - it needs to be fixed.

Current workaround is to set `QUARTO_PYTHON` - I do that in my powershell profile 
````powershell
$env:QUARTO_PYTHON=$(pyenv which python3)
````

> So, what is the actual recommendation for Windows users to install Python?

You're absolutely right that we need to sync the recommandations. 
#1463 is still to be handle and `pyenv-win` install needs to be supported on Windows by Quarto. 

> Also, why do I get the warning message? Why is R version 4.0.2 requested?

Are you using **renv** maybe ?

**Comment by cderv** [comment]
Duplicate of #1463

**Comment by aalexandersson** [comment]
> Are you using **renv** maybe ?

@cderv Yes. I have now removed the `renv` "lockfile" `renv.lock` in the project directory C:\temp, and as a result the warning message disappeared. I will follow issue 1463. Thanks for all your help! :-)

**Comment by cderv** [comment]
You're welcome. I'll close this issue but only because we'll follow in #1463. Easier to track only one issue. Thank you !

**Comment by emilyriederer** [comment]
> Current workaround is to set `QUARTO_PYTHON` - I do that in my powershell profile
> 
> ```powershell
> $env:QUARTO_PYTHON=$(pyenv which python3)
> ```

Just dropping by to say thank you so much for this -- great workaround!

**Comment by cderv** [comment]
👋 @emilyriederer ! 

Yeah, that is still the best workaround, and TBH it works quite well, but I'll still plan to improve detection for #1463

**Comment by matthew-e-brown** [comment]
> Current workaround is to set `QUARTO_PYTHON` - I do that in my powershell profile
> 
> ```powershell
> $env:QUARTO_PYTHON=$(pyenv which python3)
> ```

Just wanted to share my own workaround here, too. I want Quarto to see Python from within VS Code as well, which is not spawned through Powershell or Bash. So instead of adding it to an rc file, I put it into the Windows-level user environment variables.

Pyenv provides a shim for mapping `python` into the currently selected version of Python. Quarto successfully detects the full path from this shim:



<img width="500" src="https://github.com/user-attachments/assets/48469f07-1e3b-42b6-8f3f-e3afdbac7fc0">  
<img width="500" src="https://github.com/user-attachments/assets/5dc61932-5c32-4401-b435-05a59ac6ae94">

**Comment by mcanouil** [comment]
> I want Quarto to see Python from within VS Code as well, which is not spawned through Powershell or Bash.

That is controlled by the Python VSCode extension. See the setting regarding interpreter path.
For proper Python discovery to work, the setting should be the default value which is "Python". If yours contain a path, then that is why Quarto and more generally VSCode does not identify the currently active environment.
